### PR TITLE
Add group field to Diagnostics

### PR DESCRIPTION
**Pull request by sebastiencs**
See https://github.com/rust-lang-nursery/rls/pull/681

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -135,6 +135,10 @@ pub struct Diagnostic {
 
     /// The diagnostic's message.
     pub message: String,
+
+    /// A number used to associate children diagnostics with their
+    /// parent. Can be omitted.
+    pub group: Option<u64>,
```
Comment (Marwes) [inline src/lib.rs]: Can you add `#[serde(skip_serializing_if "Option::is_none")]`? That way it won't even be serialized as null

File: src/lib.rs [hunk 2]
```diff
@@ -135,6 +135,10 @@ pub struct Diagnostic {
 
     /// The diagnostic's message.
     pub message: String,
+
+    /// A number used to associate children diagnostics with their
```
Comment (Marwes) [inline src/lib.rs]: Can you adding something about this being experimental/not part of the language protocol spec?

**Comment by sebastiencs** [comment]
@Marwes Alright, I've made the necessary changes

**Comment by Marwes** [comment]
I can make a release of this tonight (GMT+1).

**Comment by Marwes** [comment]
Forgot about this yesterday but it is now released as 0.30

**Comment by sebastiencs** [comment]
@Marwes Thank you

**Pull request merged**